great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1360,10 +1360,15 @@ def get_batch_data_and_markers(
         create_temp_table: bool = batch_spec.get(
             "create_temp_table", self._create_temp_table
         )
-        if isinstance(batch_spec, RuntimeQueryBatchSpec):
+        # TODO: clean this up
+        # fix this when building the batch spec or stop doing instance checks at all
```

Comment:
TL;DR: I think it's worth some minor cleanup now.

It took me a minute to track down what's going on and why. IIUC:
* `RuntimeQueryBatchSpec` has a property called `query` (hence `getattr(batch_spec, "query", None)`).
* There can be other BatchSpecs such that `batch_spec["query"] exists (hence `batch_spec.get("query")` would exist

If that understanding is right, I think this should be restructured to better reflect that. e.g. branch for RuntimeQueryBatchSpec, else branch if we access query using dict syntax.

Also the current way this is written, `if isinstance(batch_spec, RuntimeQueryBatchSpec) or query:` is equivalent to `if query:`, which adds to the difficulty parsing this.

Reply:
I agree with everything you said.
I have no plans to merge this as is I would have marked this as a draft except I need the docs tests to run.

This is a draft showing the minimum change needed to fix the issue.
I plan on making the changes described in the TODO comments before merging.